src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -113,63 +117,134 @@
 
     /**
      * Creates a FeatureDataSource backed by the provided file that uses the given codec to decode records
-     * from that file. The data source will have no name, and will look ahead the default number of bases
-     * ({@link #DEFAULT_QUERY_LOOKAHEAD_BASES}) during queries that produce cache misses.
+     * from that file. The data source will have an automatically generated name, and will look ahead the default
+     * number of bases ({@link #DEFAULT_QUERY_LOOKAHEAD_BASES}) during queries that produce cache misses.
      *
      * @param featureFile file containing Features
-     * @param codec codec with which to decode the records from featureFile
      */
-    public FeatureDataSource( final File featureFile, final FeatureCodec<T, ?> codec ) {
-        this(featureFile, codec, null, DEFAULT_QUERY_LOOKAHEAD_BASES);
+    public FeatureDataSource( final File featureFile) {
+        this(featureFile, null);
     }
 
     /**
-     * Creates a FeatureDataSource backed by the provided File that uses the provided codec to decode records
-     * from that file, and assigns this data source a logical name. We will look ahead the default number of bases
-     * ({@link #DEFAULT_QUERY_LOOKAHEAD_BASES}) during queries that produce cache misses.
+     * Creates a FeatureDataSource backed by the provided File and assigns this data source a logical name.
+     * We will look ahead the default number of bases ({@link #DEFAULT_QUERY_LOOKAHEAD_BASES}) during queries
+     * that produce cache misses.
      *
      * @param featureFile file containing Features
-     * @param codec codec with which to decode the records from featureFile
      * @param name logical name for this data source (may be null)
      */
-    public FeatureDataSource( final File featureFile, final FeatureCodec<T, ?> codec, final String name ) {
-        this(featureFile, codec, name, DEFAULT_QUERY_LOOKAHEAD_BASES);
+    public FeatureDataSource( final File featureFile, final String name ) {
+        this(featureFile, name, DEFAULT_QUERY_LOOKAHEAD_BASES);
     }
 
     /**
+     * Creates a FeatureDataSource backed by the provided File and assigns this data source a logical name.
+     *
```

Comment:
"We will look ahead the specified number of bases during queries that produce cache misses."